node/api/api.go

```diff
@@ -162,3 +184,77 @@ func WriteJSON(w http.ResponseWriter, obj interface{}) {
 func WriteSuccess(w http.ResponseWriter) {
 	w.WriteHeader(http.StatusNoContent)
 }
+
+func GetPaginationDefaultRequest(req *http.Request) PaginationRequest {
+	return GetPaginationRequest(req, "start", "limit")
+}
+
+func GetPaginationRequest(req *http.Request, startQueryParam string, limitQueryParam string) PaginationRequest {
+	startString := req.FormValue(startQueryParam)
+	if startString == "" {
+		return PaginationRequest{
+			HasPaginationQuery:     false,
+			PaginationQueryIsValid: false,
+			Start: 0,
+			Limit: 0,
+		}
+	}
+	startIndex, err := strconv.Atoi(startString)
+	if err != nil || startIndex < 0 {
+		return PaginationRequest{
+			HasPaginationQuery:     true,
+			PaginationQueryIsValid: false,
+			Start: 0,
+			Limit: 0,
+		}
+	}
+	limit := DefaultPaginationSize
+	limitString := req.FormValue(limitQueryParam)
+	if limitString != "" {
+		limit, err = strconv.Atoi(limitString)
+		if err != nil || limit <= 0 {
+			return PaginationRequest{
+				HasPaginationQuery:     true,
+				PaginationQueryIsValid: false,
+				Start: 0,
+				Limit: 0,
+			}
+		}
+	}
+	return PaginationRequest{
+		HasPaginationQuery:     true,
+		PaginationQueryIsValid: true,
+		Start: startIndex,
+		Limit: limit,
+	}
+}
+
+// returns the starting index, end index, and total pages of a given slice for the page
+func GetPaginationIndicesAndResponse(sliceInterface interface{}, paginationRequest PaginationRequest) (PaginationWrapper, PaginationResponse) {
+	if paginationRequest.Limit <= 0 {
+		build.Critical("pagination request limit must be greater than 0")
```

Comment:
`build.Critical` should be used as a last resort. In production this will only print to the terminal and will continue the execution of the program. In a debug build this panics.
You should rather return an `errors.New("pagination request limit must be greater than 0")` and write that error to the http response later.